DiscordChatExporter.Domain/Discord/Models/Channel.cs

```diff
@@ -36,44 +37,67 @@ public partial class Channel : IHasId
 
         public Snowflake GuildId { get; }
 
-        public string Category { get; }
+        public Channel? Category { get; }
```

Comment:
This should never be nullable because we have `GetDefaultCategory`

Reply:
This brings up an issue: since categories are also channels, what should the category of a default category be? Previously I had used null, as using `GetDefaultCategory` would create infinite recursion. One thought is to modify the `Channel` constructor to set a category's category to itself if none is provided, but that feels hacky.

Reply:
Good point, I didn't realize that. Raised it in a separate discussion.